### ISSUE TITLE
LXC appliance: Patch needed for /etc/init.d/lxc

### ISSUE THREAD

**Issue by Dude4Linux** [turnkeylinux/tracker#565]
I need to apply a patch to replace the workaround for Ansible can't run lxc_container I reported previously.  I'd like to use git to apply the patch during the tkldev build, i.e. conf.d/main, but I can't figure out where to put the patch file.  At times I miss TKLpatch.  Does anyone have any suggestions?
This is a bug-fix for v14.1.
```
diff --git a/init.d/lxc b/init.d/lxc
index 54e5dc2..9d7103f 100755
--- a/init.d/lxc
+++ b/init.d/lxc
@@ -105,11 +105,12 @@ case "$1" in
        fi
 
        # Start containers
+       touch "$localstatedir"/lock/subsys/lxc
        wait_for_bridge
        # Start autoboot containers first then the NULL group "onboot,".
        log_daemon_msg "Starting LXC autoboot containers: "
        "$bindir"/lxc-autostart $OPTIONS $BOOTGROUPS
-       touch "$localstatedir"/lock/subsys/lxc
+       rm -f "$localstatedir"/lock/subsys/lxc
        ;;
   stop)
        if [ -n "$SHUTDOWNDELAY" ]
```

**Comment by Dude4Linux** [comment]
A more complete patch for the problem is:
```
diff --git a/init.d/lxc b/init.d/lxc
index 54e5dc2..d067e4b 100755
--- a/init.d/lxc
+++ b/init.d/lxc
@@ -104,12 +104,13 @@ case "$1" in
 		BOOTGROUPS="-g $BOOTGROUPS"
 	fi
 
+	touch "$localstatedir"/lock/subsys/lxc
 	# Start containers
 	wait_for_bridge
 	# Start autoboot containers first then the NULL group "onboot,".
 	log_daemon_msg "Starting LXC autoboot containers: "
 	"$bindir"/lxc-autostart $OPTIONS $BOOTGROUPS
-	touch "$localstatedir"/lock/subsys/lxc
+	rm -f "$localstatedir"/lock/subsys/lxc
 	;;
   stop)
 	if [ -n "$SHUTDOWNDELAY" ]
@@ -122,7 +123,6 @@ case "$1" in
 	# parallelized...  Even 5 second timout may be too long.
 	log_daemon_msg "Stopping LXC containers: "
 	"$bindir"/lxc-autostart $STOPOPTS $SHUTDOWNDELAY
-	rm -f "$localstatedir"/lock/subsys/lxc
 	;;
   restart|reload|force-reload)
 	$0 stop
```

In addition to applying the patch, the systemd configuration needs to be re-generated in order to make the change effective.

**Comment by Dude4Linux** [comment]
I should have included some background in the initial posting.  In studying why the '/var/lock/subsys/lxc' lock always seems to be set while the lxc service is running, I discovered what I think is a bug in the Debian packaging for Jessie and later versions.

In all the versions of LXC that I've examined, in lxc/config/init/common/lxc-containers.in
```
case "$1" in
    start)
        [ "x$LXC_AUTO" = "xtrue" ] || { exit 0; }

        [ ! -f "$lockdir"/lxc ] || { exit 0; }

        if [ -n "$BOOTGROUPS" ]; then
            BOOTGROUPS="-g $BOOTGROUPS"
        fi
        touch "$lockdir"/lxc
        # Start containers
        wait_for_bridge

        # Start autoboot containers first then the NULL group "onboot,".
        "$bindir"/lxc-autostart $OPTIONS $BOOTGROUPS
        rm -f "$lockdir"/lxc
    ;;
```
the lock is applied just before the containers are started, and removed at the end of the start sequence.  In the init scripts supplied by the Debian package, the lock is applied at the end of the start sequence and removed at the end of the stop sequence.  The purpose of the above patch is to move the two statements to their proper places.

**Comment by Dude4Linux** [comment]
Bug#813838: Acknowledgement (lxc: init scripts apply & remove lock at inappropriate times) has been reported to Debian.

**Comment by Dude4Linux** [comment]
After scratching my head a good deal and reviewing TKLpatch, I can't see any good way to accomplish this.  I've decided to punt and just use an overlay.  Will just have to remember to remove it when it's no longer necessary.

**Comment by Dude4Linux** [comment]
Issued a pull request to close this issue.  See: https://github.com/turnkeylinux-apps/lxc/pull/6

**Comment by JedMeister** [comment]
Thanks heaps John. I'll make sure that this makes it into v14.1. Sorry on my radio silence. I've been working offline to try to get v14.1 out the door. It's still not done but getting close! :smile: 

Also I'm going to reopen this issue until the fix is merged. That will make sure that it doesn't get forgotten!

FWIW many of the appliances will just have updates applied via TKLPatch for v14.1. So I opened an [issue re TKLPatch](https://github.com/turnkeylinux/tracker/issues/564) and have a fix for it. Alon will merge the code and rebuild TKLPatch for the package repo but if you want to use it before then you can work around the problem by manually updating `/usr/bin/tklpatch-prepare-cdroot` - as per the code in [the PR](https://github.com/turnkeylinux/tklpatch/pull/1/files)

**Comment by JedMeister** [comment]
Closed by https://github.com/turnkeylinux-apps/lxc/pull/6

FWIW @Dude4Linux if in the initial PR comment, instead of putting 
>See: turnkeylinux/tracker#565

you put:

>closes turnkeylinux/tracker#565

then this issue would auto close when I merge the code! No biggie but streamlines things a little! :smile:
